lisa/testsuite.py

```diff
@@ -178,6 +178,7 @@ def set_status(
             if self.message:
                 message.insert(0, self.message)
             self.message = "\n".join(message)
+            self._send_result_message(self.stacktrace)
```

Comment:
1. It may cause two results sent out together. Update the logic to send only once. Either status changed, or new message coming in.
2. Please add more comments to describe the situation. The environment may found errors after test case completed. So it needs to update for message only.
3. If a test case is passed, and then found the kernel panic, how to handle it? How can we make sure the kernel panic happening in this test case, not previous one?

Reply:
I add `send` parameter for set_status. 
This situation only happens after test case run, the node can't be connected, then will check whether it has panic or not, at that time, the env is marked as bad, so it is impossible the panic caused by previous case.

Reply:
It's ok to send again, if new message to carry. We can see what happened. But the current logic may send duplicate message on same result. Try below

```python
        send_result = False
        if message:
            if isinstance(message, str):
                message = [message]
            if self.message:
                message.insert(0, self.message)
            self.message = "\n".join(message)
            send_result = True
        if self.status != new_status:
            self.status = new_status
            if new_status == TestStatus.RUNNING:
                self._timer = create_timer()
            send_result = True
        if send_result:
            self._send_result_message(self.stacktrace)
```